resolver/resolver.go

```diff
@@ -311,15 +287,9 @@ func (res *Resolver) HandleNonMesos(w dns.ResponseWriter, r *dns.Msg) {
 		// set refused
 		m.SetRcode(r, 5)
 	} else {
-
-		proto := "udp"
-		if _, ok := w.RemoteAddr().(*net.TCPAddr); ok {
```

Comment:
ok, i'll buy that

On Fri, Sep 4, 2015 at 1:48 PM, Tomás Senart <notifications@github.com>
wrote:

> In resolver/resolver.go
> <https://github.com/mesosphere/mesos-dns/pull/242#discussion_r38776585>:
>
> > @@ -311,15 +287,9 @@ func (res *Resolver) HandleNonMesos(w dns.ResponseWriter, r *dns.Msg) {
> >  		// set refused
> >  		m.SetRcode(r, 5)
> >  	} else {
> > -
> > -		proto := "udp"
> > -		if _, ok := w.RemoteAddr().(*net.TCPAddr); ok {
>
> Taking a protocol into account means that we have to create a new client
> on every request. If we think about it, clients can't possibly depend on
> the outgoing recursion transport protocol. That's entirely transparent to
> them. What I think we should do, instead, is to attempt to exchange the
> same message with both UDP and, if that fails, due to issues such as
> excessive payload size, TCP.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/mesos-dns/pull/242/files#r38776585>.
>